state-chain/pallets/cf-swapping/src/lib.rs

```diff
@@ -2059,7 +2087,12 @@ pub mod pallet {
 
 			let estimation_output = with_transaction_unchecked(|| {
 				TransactionOutcome::Rollback(
-					Self::swap_with_network_fee(input_asset, output_asset, estimation_input).ok(),
+					Self::swap_with_network_fee_for_gas(
```

Comment:
We've renamed this to `swap_with_network_fee_for_gas` presumably because it's always used to in a context where output_asset is the gas asset. If this is the case, we might as well enforce it by removing the argument and just using the GAS_ASSET constant?

Reply:
`GAS_ASSET` is still different for different chains, and we are inside the chain-agnostic swapping pallet here... There may be a way to pass "chain" as a paramter here, e.g. by making `swap_with_network_fee_for_gas` generic? Not sure if worth it though.

Reply:
In that case I would just not rename the method - it's not gas-specific.

Reply:
But it *is* gas specific, it is just that gas assets can be different depending on the chain. The reason I renamed is that it wasn't immediately clear how this function was used, and whether or not min network fee should be enforced. Once I ensured that it is only used for gas swaps, I knew not to enforce it and renamed that function to make it clear why.